utilities/src/main/java/org/craftercms/commons/aws/AwsUtils.java

```diff
@@ -59,41 +75,67 @@ public static S3TransferManager buildTransferManager(S3AsyncClient client) {
-                    CopyObjectRequest copyObjectRequest = CopyObjectRequest.builder()
-                            .sourceBucket(sourceBucket)
-                            .sourceKey(sourceKey)
-                            .destinationBucket(targetBucket)
-                            .destinationKey(targetKey)
-                            .build();
-                    CopyRequest copyRequest = CopyRequest.builder()
-                            .copyObjectRequest(copyObjectRequest)
-                            .build();
-                    Copy copy = transferManager.copy(copyRequest);
-                    CompletedCopy completedCopy = copy.completionFuture().join();
-                    logger.debug("Finished copying '{}' from '{}' to '{}' with result '{}'", path, sourceBucket, targetBucket,
-                            completedCopy.response().copyObjectResult());
-                }, threadPoolExecutor));
+                futures.add(copyObject(sourceBucket, sourceBaseKey, targetBucket, targetBaseKey, path, transferManager, errorHandler));
             }
             CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
             logger.debug("Finished copying {} objects from '{}' to '{}'", paths.size(), sourceBucket, targetBucket);
-        } finally {
-            transferManager.close();
         }
     }
 
+    /**
+     * Convenience handler to ignore NoSuchKeyException. It will throw a CompletionException for any other exception type
+     *
+     * @return handler that ignores NoSuchKeyException
+     */
+    @SuppressWarnings("unused")
+    public static Consumer<Throwable> ignoreMissingObject() {
+        return e -> {
+            if (!(e instanceof NoSuchKeyException)) {
+                throw new CompletionException(e);
+            }
+        };
+    }
+
+    private static CompletableFuture<Void> copyObject(final String sourceBucket, final String sourceBaseKey,
+                                                      final String targetBucket, final String targetBaseKey,
+                                                      final String path, final S3TransferManager transferManager,
+                                                      final Consumer<Throwable> errorHandler) {
+        logger.debug("Copying '{}' from '{}' to '{}'", path, sourceBucket, targetBucket);
+        String sourceKey = s3KeyFromPath(sourceBaseKey, path);
+        String targetKey = s3KeyFromPath(targetBaseKey, path);
+        CopyObjectRequest copyObjectRequest = CopyObjectRequest.builder()
+                .sourceBucket(sourceBucket)
+                .sourceKey(sourceKey)
+                .destinationBucket(targetBucket)
+                .destinationKey(targetKey)
+                .build();
+        CopyRequest copyRequest = CopyRequest.builder()
+                .copyObjectRequest(copyObjectRequest)
+                .build();
+        Copy copy = transferManager.copy(copyRequest);
+        return copy.completionFuture()
```

Comment:
We had an issue with Deployer when submitting a lot of `CompletableFutures` caused a bunch of exceptions: https://github.com/craftercms/craftercms/issues/7393. Should we use the `threadPoolExecutor` here too?

Reply:
@avasquez614 we are using it. In line 85 we build the TransferManager using the `threadPoolExecutor`